pkgs/stdenv/generic/setup.sh

```diff
@@ -383,10 +383,12 @@ printWords() {
 ######################################################################
 # Initialisation.
 
-# export all vars that should be in the ENV
-for envVar in "${!env[@]}"; do
-    declare -x "${envVar}=${env[${envVar}]}"
-done
+if [ -n "$__structuredAttrs" ]; then
+    # export all vars that should be in the ENV
+    for envVar in "${!env[@]}"; do
+        declare -x "${envVar}=${env[${envVar}]}"
+    done
+fi
```

Comment:
```suggestion
# if structuredAttrs then export all vars that should be in the exported ENV
# only required with structuredAttrs because without it all attrs are already exported
if [[ -n "$__structuredAttrs" ]]; then
    for envVar in "${!env[@]}"; do
        declare -x "${envVar}=${env[${envVar}]}"
    done
fi
```